src/runtime/HalideRuntime.h

```diff
@@ -467,13 +467,16 @@ extern int halide_error_debug_to_file_failed(void *user_context, const char *fun
                                              const char *filename, int error_code);
 // @}
 
-
 /** Types in the halide type system. They can be ints, unsigned ints,
  * or floats (of various bit-widths), or a handle (which is always pointer-sized).
  * Note that the int/uint/float values do not imply a specific bit width
  * (the bit width is expected to be encoded in a separate value).
  */
-typedef enum halide_type_code_t {
+typedef enum halide_type_code_t 
+#if __cplusplus >= 201103L
+: uint8_t
```

Comment:
Wouldn't that require halide_type_code_t::halide_type_int? That seems pretty verbose.